src/components/routes/PrivateRoute.js

```diff
@@ -0,0 +1,32 @@
+import { Redirect, Route } from 'react-router-dom';
+import React from 'react';
+import decode from 'jwt-decode';
+
+export const PrivateRoute = ({ component: Component, ...props }) => {
+  const authenticate = token => {
+    try {
+      const decoded_token = decode(token);
+      if (decoded_token.exp > (Date.now()) / 1000) {
+        console.log('valid token');
```

Comment:
Well done @samuelbwambale. Kindly remove the `console.log` statements.

Reply:
@kakaemma The console statements were erroneously pushed. Deleted.